File: ATEC.PM.Shared/SystemProjects.cs
namespace ATEC.PM.Shared;

/// <summary>
/// Commesse di sistema (contenitori, non operative).
/// </summary>
public static class SystemProjects
{
    /// <summary>Commessa fittizia per lavorazioni generiche non legate a una commessa reale.</summary>
    public const string InternaCode = "INTERNA";

    public const string InternaTitle = "Lavorazioni interne generiche";

    public const string InternaNotes = "[SYSTEM] Commessa contenitore lavorazioni generiche";

    /// <summary>
    /// Partita IVA fittizia del cliente tecnico «ATEC — Sistema», che esiste solo perché
    /// la commessa INTERNA ha bisogno di un cliente. Non è un cliente vero: va tenuto
    /// fuori dall'anagrafica.
    /// </summary>
    public const string SystemCustomerVat = "__SYSTEM_INTERNA__";

    public static bool IsSystemCode(string? code) =>
        string.Equals(code?.Trim(), InternaCode, StringComparison.OrdinalIgnoreCase);
}

File: atec-pm-web/src/lib/system-projects.ts
/** Codice della commessa di sistema per lavorazioni generiche (non operative). */
export const SYSTEM_PROJECT_INTERNA = "INTERNA"

export function isSystemProjectCode(code: string | null | undefined): boolean {
  return (code ?? "").trim().toUpperCase() === SYSTEM_PROJECT_INTERNA
}

/** Etichetta corta in griglia Lavorazioni. */
export function formatWorkRequestProjectLabel(
  code: string | null | undefined,
  title: string | null | undefined
): string {
  if (isSystemProjectCode(code)) return "INTERNA — Generica"
  const c = (code ?? "").trim()
  const t = (title ?? "").trim()
  if (c && t) return `${c} — ${t}`
  return c || t || "—"
}
